### PR TITLE
Completed Google Form script generator

### PR DESCRIPTION
**Pull request by esfoobar-usds**
This Google Form Script can be dropped on any Google Drive and run to generate the Justice 40 ingestion Form.

Pending work: assign a Google Sheet by default.

### CODE REVIEW

**Comment by lucasmbrown-usds** [comment]
This is the same issue we were talking about earlier today, but I get nervous about having to maintain duplicate copies of the same schema. As I see it now, we've got one copy of the schema in the `data_set_description_schema.yaml`, and then one copy is spread throughout the source code (e.g. where we write the options such as "Esri Shapefile"). 

Is it possible to autogenerate those options from the schema itself, so we don't have to update the options in two places everytime there's a change, we just run a script? 

Thanks!

**Comment by esfoobar-usds** [comment]
> This is the same issue we were talking about earlier today, but I get nervous about having to maintain duplicate copies of the same schema. As I see it now, we've got one copy of the schema in the `data_set_description_schema.yaml`, and then one copy is spread throughout the source code (e.g. where we write the options such as "Esri Shapefile").

> Is it possible to autogenerate those options from the schema itself, so we don't have to update the options in two places everytime there's a change, we just run a script?

Yes. I mentioned on the meeting that we would do this initial phase first of having the script file on the repo so that we can have the form by Monday and then next week start to work on a Python script to generate the Google Script programatically from the YAML definition file.

**Comment by lucasmbrown-usds** [comment]
> Yes. I mentioned on the meeting that we would do this initial phase first of having the script file on the repo so that we can have the form by Monday and then next week start to work on a Python script to generate the Google Script programatically from the YAML definition file.

Ok. I'm also a little nervous that this is only a create script; after it's run the very first time, what is our plan for how we'll run this again? Are we able to use create or update functionality so that you can update the same form "in place", or are we planning to distribute a new link each time the form is updated (perhaps obscured behind a link shortener or something)? 

Thanks! Overall looking good!

**Comment by esfoobar-usds** [comment]
> Ok. I'm also a little nervous that this is only a create script; after it's run the very first time, what is our plan for how we'll run this again? Are we able to use create or update functionality so that you can update the same form "in place", or are we planning to distribute a new link each time the form is updated (perhaps obscured behind a link shortener or something)?

[Per the caveats here](https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/39#issuecomment-846159519), there is no way that I've found where the script run can update the same form or point to the same sheet, so these are factors that push me to believe we're going to run into issues pretty quickly, unless we commit to creating a proper submission app.

**Comment by esfoobar-usds** [comment]
Hi all, I have updated the script with the following changes:

- The script now generates one single Google Form and updates it
- The script writes to one single Google Sheet, although it creates a new Sheet (tab) every time the script is run
- Added descriptions to all fields
- Added instructions on Google Sheet

You can q/a the new form [here](https://docs.google.com/forms/d/e/1FAIpQLSe9EvOe90OWyrBVGo5QbCQX9Orz3lOjWi9rgBAewUQUb6PFoA/viewform).

File: data-roadmap/data_roadmap/data_set_description_schema.yaml [hunk 1]
```diff
@@ -2,23 +2,26 @@
...
 relevance_to_environmental_justice: str(required=False)
-data_formats: enum('GeoJSON', 'Esri Shapefile (SHP, DBF, SHX)', 'GML', 'KML/KMZ',
-  'GPX', 'CSV/XLSX', 'GDB', 'MBTILES', 'LAS', required=True)
-spatial_resolution: enum('State/territory', 'County', 'Zip code', 'Census tract',
-  'Census block group', 'Exact address or lat/long', 'Other', required=True)
-public_status: enum('Not Released', 'Public', 'Public for certain audiences', 'Other',
-  required=True)
-sponsor: str(required=True)
+data_formats:
+  enum('GeoJSON', 'Esri Shapefile (SHP, DBF, SHX)', 'GML', 'KML/KMZ',
```
Comment (lucasmbrown-usds) [inline data-roadmap/data_roadmap/data_set_description_schema.yaml:6]: Nice cleanup!

File: data-roadmap/data_roadmap/data_set_descriptions/google_form_gen.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
 function createForm() {
+  var form = FormApp.openByUrl(
+    "https://docs.google.com/forms/d/1Kb9Hl82I9kPJhqS_3bQDbWWvzCxPDTZBaI7_Eo6339E/edit"
```
Comment (lucasmbrown-usds) [inline data-roadmap/data_roadmap/data_set_descriptions/google_form_gen.js:3]: Very nice!

**Comment by ginabeena** [comment]
Re: schema -- we could easily have a [type provider](https://docs.microsoft.com/en-us/dotnet/fsharp/tutorials/type-providers/) or [protobuf](https://developers.google.com/protocol-buffers) format and hook host the schema registry. Would be a high value add early on that will amortize work and provide high ROI. That way you don't have stringly type inconsistencies, you can version it, you can do some binary lensing into header values, etc.

File: data-roadmap/data_roadmap/data_set_description_field_descriptions.yaml [hunk 3]
```diff
@@ -1,39 +1,23 @@
 # There is no method for adding field descriptions to `yamale` schemas.
 # Therefore, we've created a dictionary here of fields and their descriptions.
-name: A short name of the data set.
```
Comment (switzersc-usds) [inline data-roadmap/data_roadmap/data_set_description_field_descriptions.yaml:3]: We've ended up updating the descriptions directly in the Google spreadsheet. We should update this yaml file with those descriptions.

File: data-roadmap/data_roadmap/data_set_description_schema.yaml (hunk 1 above)
Comment (switzersc-usds) [reply data-roadmap/data_roadmap/data_set_description_schema.yaml:6]: We need to add data formats to this, eg TXT. Maybe this shouldn't be an enum yet.

**Review by switzersc-usds — CHANGES_REQUESTED** [review]
We should update the YAML field descriptions with the latest from the sheet.

**Comment by switzersc-usds** [comment]
I will look into this a bit more in depth before sprint planning to see what is left here and how to update the card.

**Comment by switzersc-usds** [comment]
Closing this as we have removed the data-roadmap directory for now. We can revisit this feature set at a later date.

**Pull request closed without merging**